src/Handler/Project/Create/CreateProjectHandler.php

```diff
@@ -0,0 +1,190 @@
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Shared\Driver\Exception\Exception;
+
+final class CreateProjectHandler implements DriverCommandHandlerInterface
+{
+    public const ENABLED_SERVICES_FOR_PROJECT = [
+        GCPServiceIds::SERVICE_USAGE_SERVICE,
+        GCPServiceIds::IAM_SERVICE,
+        GCPServiceIds::BIGQUERY_SERVICE,
+    ];
+
+    public const PRIVATE_KEY_TYPE = 'TYPE_GOOGLE_CREDENTIALS_FILE';
+
+    public GCPClientManager $clientManager;
+
+    public function __construct(GCPClientManager $clientManager)
+    {
+        $this->clientManager = $clientManager;
+    }
+
+    /**
+     * @throws ValidationException
+     * @throws Exception
+     * @throws ApiException
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof CreateProjectCommand);
+
+        $nameGenerator = new NameGenerator($command->getStackPrefix());
+
+        $projectId = $nameGenerator->createProjectId($command->getProjectId());
+
+        $meta = $credentials->getMeta();
+        if ($meta !== null) {
+            // override root user and use other database as root
+            $meta = $meta->unpack();
+            assert($meta instanceof GenericBackendCredentials\BigQueryCredentialsMeta);
+            $folderId = $meta->getFolderId();
+        } else {
+            throw new Exception('BigQueryCredentialsMeta is required.');
+        }
+
+        $projectsClient = $this->clientManager->getProjectClient($credentials);
+
+        $projectCreateResult = $this->createProject($projectsClient, $folderId, $projectId);
+        $projectName = $projectCreateResult->getName();
+
+        $serviceUsageClient = $this->clientManager->getServiceUsageClient($credentials);
+        $this->enableServicesForProject($serviceUsageClient, $projectName);
+
+        $projectServiceAccountId = $nameGenerator->createProjectServiceAccountId($command->getProjectId());
+        $iAmClient = $this->clientManager->getIamClient($credentials);
+        $projectServiceAccount = $this->createServiceAccount($iAmClient, $projectServiceAccountId, $projectName);
+
+        $cloudResourceManager = $this->clientManager->getCloudResourceManager($credentials);
+        $this->setPermissionsToServiceAccount($cloudResourceManager, $projectName, $projectServiceAccount->getEmail());
+        $keyFileCredentials = $this->createKeyFileCredentials($iAmClient, $projectServiceAccount);
+
+        $meta = new \Google\Protobuf\Any();
+        $meta->pack(
+            (new CreateProjectResponse\CreateProjectBigQueryResponseMeta())
+                ->setCredentials($keyFileCredentials)
+        );
+        return (new CreateProjectResponse())->setMeta($meta);
```

Comment:
Když to vidím tak přemýšlám jak to naroubovat na to naše API. Credentials normálně posíláme jako user a secret tak jestli bysme to neměli rozdělit aj tady. Pak by to bylo transparentní aj s ukládáním do metadat, prostě sa uloží user a secret, který to samo zašifruje, takto to bude mět všude custom handling.